google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -52,14 +54,15 @@
   private static final Duration MAX_CHANNEL_RECONNECT_BACKOFF = Duration.ofSeconds(10);
   private static final int MAX_PER_REQUEST_CHANGES = 10000;
 
-  private final AtomicLong channelReconnectBackoffMillis =
-      new AtomicLong(INITIAL_CHANNEL_RECONNECT_BACKOFF.toMillis());
-
   private final SubscriberStub asyncStub;
-
   private final String subscription;
   private final ScheduledExecutorService executor;
   private final MessageDispatcher messageDispatcher;
+
+  private final AtomicLong channelReconnectBackoffMillis =
+      new AtomicLong(INITIAL_CHANNEL_RECONNECT_BACKOFF.toMillis());
+
+  private final Lock lock = new ReentrantLock();
```

Comment:
I think in general we prefer ReadWriteLock to ReentrantLock, it should be possible to guarantee we won't have re-entry, and I think it's generally preferred to disallow it in order to simplify the concurrency model.

If you feel strongly and/or this is used widely elsewhere feel free to disregard.

Reply:
Ah okay, fair enough, based on name alone I thought it may be re-entrant specifically.